### PR TITLE
fix: collect fees extrinsics 

### LINKED ISSUE #544: Fixes required in collect fees pallet
- [x] Remove for loop from [here](https://github.com/Polkadex-Substrate/Polkadex/blob/f299d744deabb7830d97347d0ad8e48387661242/pallets/ocex/src/lib.rs#L695) as some one may do a double spend attack .
- [x] Clean the fees vec after coming out of loop

### PR DESCRIPTION
**Pull request by felixfaisal**
This PR resolves #544

### CODE REVIEW

**Review by Gauthamastro — DISMISSED** [review]

File: pallets/ocex/src/lib.rs [hunk 1]
```diff
@@ -691,22 +691,37 @@ pub mod pallet {
...
+									&Self::get_custodian_account(),
+									&beneficiary,
+									converted_fee.saturated_into(),
+									fees.asset,
+								)
+								.is_err()
+								{
+									// Push it back inside the internal vector
+									internal_vector.try_push(fees).unwrap_or_default();
+									break
```
Comment (serhii-temchenko) [inline pallets/ocex/src/lib.rs]: Not really sure why it is `Ok` result if it's unable to transfer asset for example just in the middle of collection iteration.

File: pallets/ocex/src/lib.rs [hunk 2]
```diff
@@ -691,22 +691,37 @@ pub mod pallet {
...
+								internal_vector.try_push(fees).unwrap_or_default();
+								return Err(Error::<T>::FailedToConvertDecimaltoBalance)
+							}
+						}
+					}
+					return Ok(())
+				})
+				.is_ok(),
+				Error::<T>::FailedToConvertDecimaltoBalance
+			);
```
Comment (serhii-temchenko) [inline pallets/ocex/src/lib.rs:764]: No tests refactoring for new implementation - I assume there's no tests or not really good tests if not failing in this situation.

**Review by serhii-temchenko — CHANGES_REQUESTED** [review]

File: pallets/ocex/src/lib.rs (hunk 2 above)
Comment (felixfaisal) [reply pallets/ocex/src/lib.rs:764]: We need to write a test for this situation

File: pallets/ocex/src/lib.rs (hunk 1 above)
Comment (felixfaisal) [reply pallets/ocex/src/lib.rs]: I didn't think this through, I can return an `Err` here

Comment (felixfaisal) [reply pallets/ocex/src/lib.rs]: Updated ✅

File: pallets/ocex/src/lib.rs (hunk 2 above)
Comment (felixfaisal) [reply pallets/ocex/src/lib.rs:764]: Updated ✅

**Review by serhii-temchenko — APPROVED** [review]

**Review by Gauthamastro — APPROVED** [review]

**Pull request merged**